python_modules/dagster/dagster/core/definitions/job_definition.py

```diff
@@ -251,6 +252,16 @@ def get_partition_set_def(self) -> Optional["PartitionSetDefinition"]:
 
         return self._cached_partition_set
 
+    def run_request_for_partition(self, run_key: Optional[str], partition_key: str) -> RunRequest:
```

Comment:
Nits:
* Can `run_key` get a default value so users don't need to specify it?
* Given that it will only make sense to supply `run_key` sometimes, can `partition_key` come first so users can supply it by itself as a non-keyword arg?

Reply:
I think we should mirror `RunRequest` w.r.t `run_key` behavior.

Choosing to supply run_key or not has pretty big implications on the default idempotency, especially for sensors, so I think that should be a separate issue / discussion.

I think it makes sense to swap the order though.

Reply:
Got it - makes sense.